### DISCUSSION TITLE
Migrating from Cruft where not all projects are up to date

### DISCUSSION THREAD

**Discussion by slocoro** [copier-org/copier#1637] [Q&A]
Hi,

We're considering migrating a template currently maintained wtih Cruft to Copier. We plan on using [this approach](https://github.com/orgs/copier-org/discussions/1393) to migrate projects from Cruft to Copier.

However, I'm not sure to do about projects that are not up to date with the latest version of the Cruft template. Assume we have two versions of the Cruft template, v1 and v2. Some users have a project created from v1, others from v2, and others have updated their project from v1 to v2. For users that are on v2, the migration to Copier will be easy. But what about users that still have a project based on the V1 of the template? If we migrate those projects to the Copier template (that reflects v2) users will not be able to get the updates introduced from v1 to v2.

Assuming we can't force all users to updrade their projects to v2 before migrating to Copier. Is the only way we can enable updates for those users to retag the Copier template with a new version v3 (without having made changes to it)?

**Comment by yajo** [comment]
I guess you'll need to create a Copier template v1 and develop a migration process for v2. Make sure to add git tags for both.

Then tell your users to add `-r v1` when migrating to copier if their project is stll on V1.
